lib/api/route/realm.dart

```diff
@@ -0,0 +1,67 @@
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+import 'package:json_annotation/json_annotation.dart';
+
+part 'realm.g.dart';
+
+/// https://zulip.com/api/get-server-settings
+///
+/// Despite the name, this is really a home for certain realm-specific
+/// settings, as well as some information about the server as a whole.
+Future<GetServerSettingsResult> getServerSettings({
+  required String realmUrl,
+}) async {
+  // TODO dedupe this part with LiveApiConnection; make this function testable
+  final response = await http.get(
+    Uri.parse("$realmUrl/api/v1/server_settings"));
+  if (response.statusCode != 200) {
+    throw Exception('error on GET server_settings: status ${response.statusCode}');
+  }
+  final data = utf8.decode(response.bodyBytes);
+
+  final json = jsonDecode(data);
+  return GetServerSettingsResult.fromJson(json);
+}
+
+@JsonSerializable(fieldRename: FieldRename.snake)
+class GetServerSettingsResult {
```

Comment:
zulip-mobile has this comment:

```js
// Current to FL 121.
```

Should we put a similar comment here? No harm in copying zulip-mobile with 121, I think, since callers don't yet use newer server features.

Reply:
I'd like to use those sparingly. In general I'd prefer that the expectation is that things are up to date.

On this get-server-settings endpoint, for example, now that the current feature level is up to 171, that comment makes it look like it might be pretty stale; but in reality the last change it had (*) was at FL 116, so it's still perfectly up to date. I think most of the Zulip API is like this endpoint in that it doesn't change often, and we can pretty well keep it up to date.

The major exception is settings, which the Git history on zulip-mobile reminds me is where we first started applying this "current to" convention in zulip/zulip-mobile@3971c9c47. I think it may well make sense to use it in a couple of spots that have long lists of settings, once we add those here.

One practice we could start in order to systematically catch anything that's out of date: on each server release, take that as a prompt/reminder to go through the API changelog since the last release, find all the updates that are to API surface we describe, and apply those updates where we haven't already. Looking back at [the 5.0-to-6.0 changes](https://zulip.com/api/changelog#changes-in-zulip-60) as an example, I think that'd be a quite manageable task.

(*) Apart from the `ignored_parameters_unsupported` thing, which is cross-cutting over the whole API and which we're not interested in anyway.

Reply:
Ah OK, that makes sense.